### PR TITLE
Samples: Fix XEC clock control sample code for MEC172x using new GPIOs.

### PR DESCRIPTION
**Pull request by jjdalynh**
This commit goes along with pull request #61396 to update the sample code to use simplified GPIO SOC definitions

### CODE REVIEW

**Comment by jjdalynh** [comment]
@MaureenHelm @galak @ioannisg this PR will fix the build issue for PR #61396 (if there was a way to specify dependent PRs i would have)

File: samples/drivers/clock_control_xec/src/main.c [hunk 1]
```diff
@@ -179,8 +181,10 @@ static void vbat_power_fail(void)
 }
 #endif
 
-static const struct gpio_ctrl_regs * const gpc =
-	(struct gpio_ctrl_regs *)(DT_REG_ADDR(DT_NODELABEL(gpio_000_036)));
+#define MCHP_XEC_PINCFG(pin) ((pin >> 5) * 32 + (pin & 0x1f))
```
Comment (jvasanth1) [inline samples/drivers/clock_control_xec/src/main.c:184]: can we move this macro to the soc layer?

**Comment by Manimaran-A** [comment]
@jjdalynh In file "soc\arm\microchip_mec\mec172x\soc_power_debug.h"

We are using the enum "MCHP_GPIO_0241_ID"  as below
#define DP_GPIO_ID		MCHP_GPIO_0241_ID

Above line also needs modification per our latest logic.

**Comment by jjdalynh** [comment]
> 

I intend to modify this code in a separate commit.  This code is very platform-specific and doesn't belong under soc/arm/microchip_mec/* in any way.  My thoughts are that the macro needs to point to a weak function and wrap everything with a CONFIG_SOC_SLEEP_STATE_DEBUG_FUNCTION in order to provide a mechanism for boards to provide a platform-specific function rather than wrapping it with the #define SOC_SLEEP_STATE_GPIO_MARKER_DEBUG at the top of the file.

**Comment by Manimaran-A** [comment]
> CONFIG_SOC_SLEEP_STATE_DEBUG_FUNCTION

Ok

File: samples/drivers/clock_control_xec/src/main.c (hunk 1 above)
Comment (jjdalynh) [reply samples/drivers/clock_control_xec/src/main.c]: done.

**Review by jvasanth1 — APPROVED** [review]

**Review by albertofloyd — APPROVED** [review]

**Pull request merged**